src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromDevider = fromFormat[3];
+  const toDevider = toFormat[3];
+
+  const dateValues = date.split(fromDevider);
+
+  const objectDate = {};
+
+  fromFormat.forEach((key, value) => {
+    objectDate[key] = dateValues[value];
+  });
+
+  if (toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(2);
+  }
+
+  if (objectDate['YY'] && toFormat.includes('YYYY')) {
```

Comment:
The condition should check if 'YYYY' exists in 'objectDate' before trying to access 'objectDate['YY']'.